test/pool.test.ts

```diff
@@ -28,6 +28,41 @@ async function mineBlocksUntilCycleEnd(): Promise<void> {
 
 type AnyPool = EthPool | Erc20Pool;
 
+class ProxyReceiverWeight {
+  receiverWeight: number;
+  proxyWeight: number;
+
+  constructor(receiverWeight: number, proxyWeight: number) {
+    this.receiverWeight = receiverWeight;
+    this.proxyWeight = proxyWeight;
+  }
+
+  static empty(): ProxyReceiverWeight {
+    return new ProxyReceiverWeight(0, 0);
+  }
+
+  isEmpty(): boolean {
+    return this.receiverWeight == 0 && this.proxyWeight == 0;
+  }
+}
+
+type ProxyReceiverWeights = Map<string, ProxyReceiverWeight>;
+
+function updateReceiversList(
```

Comment:
```suggestion
function updateReceiverWeights(
```
This would align the noun in the function name with the type of the subject and make it clearer what is going on. It also confused me that we were using the term “list” whereas the actual type is a map.

Reply:
Renamed.

This might be a mistake, but this data structure is called a `list` consistently throughout the codebase. The underlying data structure is a hybrid between a map and a linked list. I agree that `map` would make slightly more sense, because its API resembles a classic map with its free key-based access. If that's confusing, I can rename it, but in a separate PR. What do you think?

Reply:
Let’s leave it like it is. There’s no clear case for either “list” or “map”.